src/syntax/Scanner.js

```diff
@@ -527,17 +527,31 @@ function peekTokenNoLineTerminator() {
   let end = t.location.start.offset;
   for (let i = start; i < end; i++) {
     let code = input.charCodeAt(i);
-    if (isLineTerminator(code))
+    if (isLineTerminator(code)) {
       return null;
+    }
 
-    // If we have a block comment we need to skip it since new lines inside
-    // the comment are not significant.
+    // If a block comment contains new lines then the comment is treated as a
+    // new line.
     if (code === 47) {  // '/'
       code = input.charCodeAt(++i);
       // End of line comments always mean a new line is present.
-      if (code === 47)  // '/'
+      if (code === 47) {  // '/'
         return null;
-      i = input.indexOf('*/', i) + 2;
+      }
+
```

Comment:
That makes me realize that this can be simplified further. All we need to look for is a newline since either:

 - There is a line break in whitespace
 - There is a `//` comment, in which case we have a new line at the end of the comment.
 - If a multiline comment and it contains a new line treat the comment as a newline.